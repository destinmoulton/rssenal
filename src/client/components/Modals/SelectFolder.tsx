import * as React from "react";

import { Dropdown } from "semantic-ui-react";

import { TFolders, TFolderID } from "../../types";

interface ISelectFolderProps {
    onChange: (folderId: TFolderID) => void;
    selectedValue: string;
    folders: TFolders;
}

class SelectFolder extends React.Component<ISelectFolderProps> {
    _handleOnChange = (e: any, dropdown: any) => {
        const selectedValue = dropdown.value;

        this.props.onChange(selectedValue);
    };

    render() {
        const { folders, selectedValue } = this.props;

        // Convert the feed folders Immutable list into
        // an array of key/value options.
        let folderOptions = folders
            .map(folder => {
                return {
                    key: folder._id,
                    value: folder._id,
                    text: folder.name
                };
            })
            .toArray();

        // Put "Uncategorized" first
        folderOptions.unshift(folderOptions.pop());

        return (
            <Dropdown
                value={selectedValue}
                options={folderOptions}
                onChange={this._handleOnChange}
                selection
            />
        );
    }
}

export default SelectFolder;
