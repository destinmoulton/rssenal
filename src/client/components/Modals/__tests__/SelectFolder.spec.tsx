import * as React from "react";

import { shallow } from "enzyme";

import SelectFolder from "../SelectFolder";
import folder from "../../../../../test/data/folder";
import folders from "../../../../../test/data/immutable/folders";

describe("<SelectFolder />", () => {
    it("renders and matches the snapshot", () => {
        const wrapper = shallow(
            <SelectFolder
                folders={folders}
                selectedValue={folder._id}
                onChange={jest.fn()}
            />
        );
        expect(wrapper).toMatchSnapshot();
    });
});
