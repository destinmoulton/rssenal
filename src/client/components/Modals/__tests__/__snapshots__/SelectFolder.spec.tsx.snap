// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SelectFolder /> renders and matches the snapshot 1`] = `
<Dropdown
  additionLabel="Add "
  additionPosition="top"
  closeOnBlur={true}
  deburr={false}
  icon="dropdown"
  minCharacters={1}
  noResultsMessage="No results found."
  onChange={[Function]}
  openOnFocus={true}
  options={
    Array [
      Object {
        "key": "5b33c57eb2438d5708dc1968",
        "text": "Programming",
        "value": "5b33c57eb2438d5708dc1968",
      },
      Object {
        "key": "5b33c4bab2438d5708dc1967",
        "text": "Random",
        "value": "5b33c4bab2438d5708dc1967",
      },
    ]
  }
  renderLabel={[Function]}
  searchInput="text"
  selectOnBlur={true}
  selectOnNavigation={true}
  selection={true}
  value="5b33c4bab2438d5708dc1967"
  wrapSelection={true}
/>
`;
